ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Allocation.scala

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.infrastructure
+
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.digitalasset.ledger.client.binding.Primitive.Party
+
+private[testtool] object Allocation {
+  def allocate(partyCounts: PartyCount*): ParticipantAllocation =
+    ParticipantAllocation(partyCounts)
+
+  final case class ParticipantAllocation(partyCounts: Seq[PartyCount])
+
+  sealed trait PartyCount {
+    val count: Int
+  }
+
+  case object NoParties extends PartyCount {
+    override val count = 0
+  }
+
+  case object SingleParty extends PartyCount {
+    override val count = 1
+  }
+
+  case object TwoParties extends PartyCount {
+    override val count = 2
+  }
+
+  final case class Parties(override val count: Int) extends PartyCount
+
+  final case class Participants(participants: Participant*)
```

Comment:
If you make this constructor private and you make your own that takes a participant and then a vararg of participants you disallow empty `Participants` (which we probably want).

Reply:
I made it private to the `infrastructure` package and changed `allocate` to take at least one element, which should provide the safety we're looking for.